### ISSUE TITLE
Feature Request: Export a helper function that lists all the internal packages

### ISSUE THREAD

**Issue by BenJackGill** [0x80/isolate-package#104]
It would be great if you could provide a function that returned all the internal package names as a string array.

This would make integrating `isolate-package` and `tsup` much simpler.

At the moment we have to manually list all of the internal packages in the tsup `noExternal` config property. But sometimes I forget to do that and I see strange errors that are hard to track down until I remember to add any new internal packages to the `noExternal` list.

I assume `isolate-package` has a way of identifying the internal packages, so if you could export that as a helper function for us we could easily use it within tsup to solve this issue.

This is how my current **tsup.config.ts** file looks. I have to manually list all the internal packages. This is error prone because I often forget to list them here. 
```TypeScript
import { defineConfig } from "tsup";

export default defineConfig({
  clean: true,
  entry: ["src/**/*.ts"],
  format: ["esm"],
  // List all internal packages here
  noExternal: [
    "@repo/firebase-functions-helpers",
    "@repo/firestore-converter",
    "@repo/types",
    "@repo/tsconfig",
    "@repo/utils",
  ],
  sourcemap: true,
  treeshake: true,
  tsconfig: "tsconfig.json",
});
```

This is how is could it work if you exported a helper function for us. The internal packages are automatically listed using an `isolate-package` helper function named `internalPackages` (just a name suggestion).
```TypeScript
import { defineConfig } from "tsup";
import { internalPackages } from "isolate-package"; // <--- Possible helper function exported from isolate-package

export default defineConfig({
  clean: true,
  entry: ["src/**/*.ts"],
  format: ["esm"],
  // List all internal packages here
  noExternal: internalPackages(),
  sourcemap: true,
  treeshake: true,
  tsconfig: "tsconfig.json",
});
```

**Comment by 0x80** [comment]
What happens if you don't list the packages? 

I am using `tsup` and I've never had to list my packages under noExternal. AFAIK it's perfectly fine to treat the packages as external, because the manifest points to the local directories within the isolated output.

**Comment by 0x80** [comment]
Ah yes that makes sense! 

I think it's a good suggestion. It should be easy to add, but I'm not sure when I'll get to it. Feel free to submit a PR ;)

**Comment by BenJackGill** [comment]
I tried to attempt a PR today for this but I'm afraid I'm not skilled enough, sorry. Might try again when I have more time.

**Comment by 0x80** [comment]
Hi Ben,

I am tempted to help you out, but I am really swamped with work and personally I have no use for it.

When I first created mono-ts and wrote about it, there were a few things different from now:

- Turborepo didn't have a watch task
- I didn't understand that TS project references were sufficient to get go-to-definition working.
- Using incremental builds for TS seems to have helped my setup too

I've updated / simplified the mono-ts boilerplate to reflect these things. In my projects I no longer perceive the build stage as friction, and things seem to work smoothly overall.

I suspect not a lot of people are using the internal packages strategy, and as build tooling improves I think it might become less relevant.

**Comment by 0x80** [comment]
@BenJackGill I hope you can still use this...

**Comment by BenJackGill** [comment]
> [@BenJackGill](https://github.com/BenJackGill) I hope you can still use this...

Yes I can! Thanks, will try it today :) You are a saint for maintaining this repo
